tools/run_tests/run_xds_tests.py

```diff
@@ -950,6 +969,289 @@ def prepare_services_for_urlmap_tests(gcp, original_backend_service,
+        new_client_process = None
+        if client_process:
+            client_process.terminate()
+            client_cmd = shlex.split(client_cmd_formatted_no_uri.format(
+                                     server_uri=server_uri))
+            new_client_process = subprocess.Popen(client_cmd,
+                                                  env=client_env,
+                                                  stderr=subprocess.STDOUT,
+                                                  stdout=test_log_file)
+        return new_client_process
+
+
+def test_forwarding_rule_port_match(gcp, backend_service, instance_group,
+                                    client_process):
+    logger.info("Running test_forwarding_rule_port_match")
+    try:
+        backend_instances = get_instance_names(gcp, instance_group)
+        wait_until_all_rpcs_go_to_given_backends(backend_instances,
+                                                 _WAIT_FOR_STATS_SEC)
+        delete_global_forwarding_rule(gcp)
+        create_global_forwarding_rule(gcp, forwarding_rule_name,
+                                      [x for x in parse_port_range(_DEFAULT_PORT_RANGE)
+                                       if x != gcp.service_port])
+        wait_until_no_rpcs_go_to_given_backends(backend_instances,
+                                                _WAIT_FOR_STATS_SEC)
+
+        # test fr default port: omit port in client request service uri, no port in url-map
+        if client_process:
+            client_process.terminate()
+        client_cmd_formatted = client_cmd_formatted_no_uri.format(
+                server_uri=service_host_name)
+        client_cmd = shlex.split(client_cmd_formatted)
+        local_client_process = subprocess.Popen(client_cmd,
+                                                env=client_env,
+                                                stderr=subprocess.STDOUT,
+                                                stdout=test_log_file)
+        delete_global_forwarding_rule(gcp)
+        delete_target_proxy(gcp)
+        delete_url_map(gcp)
+        create_url_map(gcp, url_map_name, backend_service, service_host_name)
+        create_target_proxy(gcp, gcp.target_proxy.name, False)
+        potential_ip_addresses = []
+        max_attempts = 10
+        for i in range(max_attempts):
+            potential_ip_addresses.append('10.10.10.%d' % (random.randint(0, 255)))
+        create_global_forwarding_rule(gcp, forwarding_rule_name,
+                                      [80], potential_ip_addresses)
+        wait_until_all_rpcs_go_to_given_backends(backend_instances,
+                                                 _WAIT_FOR_STATS_SEC)
+
+        # omit port in client request uri, specify port in url-map
+        patch_url_map_host_rule_with_port(gcp, url_map_name,
+                                          backend_service, service_host_name)
+        wait_until_no_rpcs_go_to_given_backends(backend_instances,
```

Comment:
Not clear why rpcs are expected to fail: using default port everywhere (80) and client request skips port so it should work, no?

Reply:
verified that the behavior is no match, consistent with [public doc](https://cloud.google.com/traffic-director/docs/proxyless-overview#proxyless-url-map), see the last bullet point.